code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -1715,8 +1723,21 @@ func (sc *StorageSmartContract) finishAllocation(
 		Amount:       i,
 	})
 
-	alloc.Finalized = true
+	if openChallengesAvailable {
```

Comment:
you can iterate empty list the same way, do you really need `openChallengesAvailable`?

Reply:
Let me remove it.